src/main/kotlin/com/hibob/academy/employeeFeedback/service/FeedbackService.kt

```diff
@@ -0,0 +1,43 @@
+package com.hibob.academy.employeeFeedback.service
+
+import com.hibob.academy.employeeFeedback.dao.*
+import jakarta.ws.rs.BadRequestException
+import org.springframework.stereotype.Service
+
+@Service
+class FeedbackService(private val feedbackDao: FeedbackDao, private val employeeDao: EmployeeDao) {
+    fun submitFeedback(employeeId: Int, companyId: Int, anonymity: AnonymityType, feedback: String): Boolean {
+        var employeeIdForFeedback: Int? = employeeId
+        if (anonymity == AnonymityType.ANONYMOUS)
+            employeeIdForFeedback = null
+
+        if (feedback.length < 30 || feedback.length > 500)
+            return false
+
+        val feedbackForSubmission =
+            FeedbackForSubmission(employeeIdForFeedback, companyId, anonymity, feedback)
+        feedbackDao.submitFeedback(feedbackForSubmission)
+        return true
+    }
+
+    fun viewAllSubmittedFeedback(employeeId: Int, companyId: Int): List<FeedbackInfo> {
+        return feedbackDao.viewAllSubmittedFeedback(companyId)
+    }
+
+    fun viewStatusOfMyFeedback(employeeId: Int, companyId: Int, feedbackId: Int): Map<Int, Boolean> {
+        val feedbackInfo = feedbackDao.getFeedbackById(feedbackId, companyId)
+        feedbackInfo?.let{
+            val idOfEmployeeThatWroteFeedbackDao = feedbackInfo.employeeId
```

Comment:
this parameter name is not good, too long and doesn't really explain anything.
What is this id?  maybe it's the feedback's employee id?

You shouldn't use Dao in the param names, unless it's the dao itself

Reply:
It's the id of the employee that wrote the feedback
I'll change it